WalletWasabi/CoinJoin/Coordinator/Coordinator.cs

```diff
@@ -433,6 +438,26 @@ public IEnumerable<uint256> GetUnconfirmedCoinJoins()
 		}
 	}
 
+	private async void CoinVerifier_CoinBlackListedAsync(object? _, Coin coin)
+	{
+		bool bannedAtLeastOne = false;
+		foreach (var round in Rounds)
+		{
+			if (round.ContainsInput(coin.Outpoint, out var alices))
+			{
+				var outPointsToBan = alices.SelectMany(a => a.Inputs).Select(i => i.Outpoint).ToArray();
+				await UtxoReferee.BanUtxosAsync(1, DateTimeOffset.UtcNow, forceNoted: false, round.RoundId, forceBan: true, outPointsToBan).ConfigureAwait(false);
+				bannedAtLeastOne = true;
+			}
+		}
+
+		if (!bannedAtLeastOne)
+		{
+			await UtxoReferee.BanUtxosAsync(1, DateTimeOffset.UtcNow, forceNoted: false, 0, forceBan: true, coin.Outpoint).ConfigureAwait(false);
+			Logger.LogError($"Could not find alice in rounds for coin:'{coin.Outpoint}'. Banned only this coin.");
+		}
```

Comment:
Is this code-part needed to ban coins from WW1 if the coin was banned from WW2?

Reply:
Good question but WW1 code will be removed in the long term, so I would not do any effort to involve Prison here and vice versa.

Reply:
Maybe consider removing the `LogError`. It's not the best idea, but right now every `WW2.0` ban will also cause a log here unnecessarily IMO.

Reply:
Removed.

Reply:
Coins will be banned with UtxoReferee and Prison as well. No matter where the coin tried to participate.